toolchains/docker/BUILD

```diff
@@ -0,0 +1,71 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+package(default_visibility = ["//visibility:private"])
+
+load(":toolchain.bzl", "docker_toolchain")
+
+# Docker toolchain type
+toolchain_type(
+    name = "toolchain_type",
+)
+
+# Default docker toolchain that expects the 'docker' executable
+# to be in the PATH
+docker_toolchain(
+    name = "default_toolchain_impl",
+    tool_path = "docker",
+    visibility = ["//visibility:public"],
```

Comment:
can we restrict the visibility of this target?

Reply:
Nope. This is actually the rule that is being accessed by rules that use this toolchain_type. Making this private will give visibility errors

Reply:
got it, sgtm